pkg/sdk/renderer/argo/renderer.go

```diff
@@ -174,7 +179,7 @@ func (r *Renderer) Render(ctx context.Context, input *RenderInput) (*RenderOutpu
 	}
 
 	// 10. Render rootWorkflow templates
-	_, err = dedicatedRenderer.RenderTemplateSteps(ctxWithTimeout, rootWorkflow, implementation.Spec.Imports, dedicatedRenderer.inputTypeInstances, "")
+	_, err = dedicatedRenderer.RenderTemplateSteps(ctxWithTimeout, rootWorkflow, implementation.Spec.Imports, typeInstancesToInject, "")
```

Comment:
Actually I can revert this change, because as I appended additional TypeInstances to the `dedicatedRenderer.inputTypeInstances`, so it's basically the same

```suggestion
	_, err = dedicatedRenderer.RenderTemplateSteps(ctxWithTimeout, rootWorkflow, implementation.Spec.Imports, dedicatedRenderer.inputTypeInstances, "")
```